src/pydagmc/dagnav.py

```diff
@@ -60,6 +61,81 @@ def volumes(self):
     def volumes_by_id(self):
         return {v.id: v for v in self.volumes}
 
+    @property
+    def volumes_by_mat_tag(self) -> Dict[str, list[Volume]]:
+        """
+        Returns a dictionary mapping material tags (names) to lists of
+        Volume objects associated with that material.
+
+        Material tags are derived from group names like "mat:material_name".
+        The dictionary key will be "material_name". Volumes without a
+        material group are excluded.
+
+        Returns
+        -------
+        Dict[str, list[Volume]]
+            A dictionary where keys are material names (str) and values
+            are lists of Volume objects.
+        """
+        material_map: Dict[str, list[Volume]] = {}
+        for volume in self.volumes:
+            material_name = volume.material
+            if material_name is not None:
+                if material_name not in material_map:
+                    material_map[material_name] = []
+                material_map[material_name].append(volume)
+        return material_map
+
+    def get_volumes_by_material(self, material_name: str, n_suggestions: int = 3, cutoff: float = 0.6) -> list[Volume]:
+        """
+        Retrieves a list of Volume objects associated with the given material name.
+
+        If the exact material name is not found, it raises a KeyError
+        and suggests potential close matches found in the model using difflib.
```

Comment:
Best to leave implementation details out here in case we change them and forget to update the docstring.

```suggestion
        and suggests potential close matches found in the model.
```